File: models/business_time.py
from models import db


def get():
    data = db.BUSINESS_COLLECTION.find_one({}, {"_id": 0})
    return data


def update(data):
    # establish update data
    update_data = {}
    week = ["mon", "tue", "wed", "thu", "fri", "sat", "sun"]
    for day in week:
        update_data[day] = {
            "start": data[day]["start"],
            "end": data[day]["end"],
        }
    # start update
    db.BUSINESS_COLLECTION.update_one({}, {"$set": update_data})

File: static/js/business_time.js
/* global FetchData , */
const bsAPI = {
  business_time: '/api/setting/business-time',
};

async function update() {
  const setTime = {};
  const week = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun'];
  week.forEach((element) => {
    setTime[element] = {
      start: document.getElementById(`${element}Start`).value,
      end: document.getElementById(`${element}End`).value,
    };
  });
  const result = await FetchData.post(bsAPI.business_time, setTime);
  if (result.status === 403) {
    document.getElementById('inform').outerText = '無此權限';
  } else if (result.status === 400) {
    document.getElementById('inform').outerText = '格式錯誤';
  } else {
    document.getElementById('inform').outerText = '更新成功';
  }
}

window.onload = function () {
  document.getElementById('btn').addEventListener('click', update);
};
